bolt/src/BinaryFunction.h

```diff
@@ -1294,51 +1294,59 @@ class BinaryFunction {
     assert(Address >= getAddress() && Address < getAddress() + getMaxSize() &&
            "address is outside of the function");
     uint64_t Offset = Address - getAddress();
+    if (BC.isAArch64()) {
+      switch (RelType) {
+      case ELF::R_AARCH64_ABS64:
+      case ELF::R_AARCH64_ADD_ABS_LO12_NC:
+      case ELF::R_AARCH64_ADR_GOT_PAGE:
+      case ELF::R_AARCH64_ADR_PREL_LO21:
+      case ELF::R_AARCH64_ADR_PREL_PG_HI21:
+      case ELF::R_AARCH64_ADR_PREL_PG_HI21_NC:
+      case ELF::R_AARCH64_LD64_GOT_LO12_NC:
+      case ELF::R_AARCH64_LDST8_ABS_LO12_NC:
+      case ELF::R_AARCH64_LDST16_ABS_LO12_NC:
+      case ELF::R_AARCH64_LDST32_ABS_LO12_NC:
+      case ELF::R_AARCH64_LDST64_ABS_LO12_NC:
+      case ELF::R_AARCH64_LDST128_ABS_LO12_NC:
+      case ELF::R_AARCH64_TLSDESC_ADD_LO12:
+      case ELF::R_AARCH64_TLSDESC_ADR_PAGE21:
+      case ELF::R_AARCH64_TLSDESC_ADR_PREL21:
+      case ELF::R_AARCH64_TLSDESC_LD64_LO12:
+      case ELF::R_AARCH64_TLSIE_ADR_GOTTPREL_PAGE21:
+      case ELF::R_AARCH64_TLSIE_LD64_GOTTPREL_LO12_NC:
+        Relocations[Offset] =
+            Relocation{Offset, Symbol, RelType, Addend, Value};
+        return;
+      case ELF::R_AARCH64_CALL26:
+      case ELF::R_AARCH64_JUMP26:
+      case ELF::R_AARCH64_TLSDESC_CALL:
+      case ELF::R_AARCH64_TLSLE_ADD_TPREL_HI12:
+      case ELF::R_AARCH64_TLSLE_ADD_TPREL_LO12_NC:
+        return;
+      default:
+        llvm_unreachable("Unexpected AArch64 relocation type in code");
+      }
+    }
+
```

Comment:
I think it's cleaner to split this member function into two   "addRelocationX86" and  "addRelocationAArch64", and then call the appropriate one in addRelocation() that will have a single line in its new body:
  `return BC.isAArch64() ? addRelocationAArch64() : addRelocationX86();`
  
We already have so many functions with   if (X)  {lots of code; return}  else  {lots of code} that can be easily refactored in this way. One example is mapCodeSections that is entirely different depending on whether we are in relocation mode or not.

What do you think @maksfb? Any preference?

Reply:
Done, made it in Relocation.cpp style